assets/js/modules/analytics/components/common/ProfileSelect.js

```diff
@@ -34,13 +34,41 @@ import { trackEvent } from '../../../../util';
 const { useSelect, useDispatch } = Data;
 
 export default function ProfileSelect() {
-	const accountID = useSelect( ( select ) => select( STORE_NAME ).getAccountID() );
-	const propertyID = useSelect( ( select ) => select( STORE_NAME ).getPropertyID() );
-	const profiles = useSelect( ( select ) => select( STORE_NAME ).getProfiles( accountID, propertyID ) );
 	const profileID = useSelect( ( select ) => select( STORE_NAME ).getProfileID() );
-	const isLoadingAccounts = useSelect( ( select ) => select( STORE_NAME ).isDoingGetAccounts() );
-	const isLoadingProperties = useSelect( ( select ) => select( STORE_NAME ).isDoingGetProperties( accountID ) );
-	const isLoadingProfiles = useSelect( ( select ) => select( STORE_NAME ).isDoingGetProfiles( accountID, propertyID ) );
+	const hasResolvedAccounts = useSelect( ( select ) => select( STORE_NAME ).hasFinishedResolution( 'getAccounts' ) );
+
+	const {
+		accountID,
+		propertyID,
+		profiles,
+		hasStartedFetchingProperties,
+		hasResolvedProperties,
+		hasStartedFetchingProfiles,
+		hasResolvedProfiles,
+	} = useSelect( ( select ) => {
+		const data = {
+			accountID: select( STORE_NAME ).getAccountID(),
+			propertyID: select( STORE_NAME ).getPropertyID(),
+			profiles: [],
+			hasStartedFetchingProperties: false,
+			hasResolvedProperties: false,
+			hasStartedFetchingProfiles: false,
+			hasResolvedProfiles: false,
```

Comment:
In this case, wouldn't we be able to use the `isResolving` meta selector instead? Then we could have e.g. `isResolvingProperties` which I believe would do the same as `hasStartedFetchingProperties && ! hasResolverProperties`?

Reply:
Ah... yes, you are right. We can use `isResolving` here. Updated.